2020-11-17 00:20:10,447 ERROR: Exception on /index [GET] [in /Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/Users/user/PycharmProjects/microblog/app/routes.py", line 29, in index
    return render_template("index.html", title='Home Page', form=form,
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/templating.py", line 138, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/jinja2/environment.py", line 930, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/jinja2/environment.py", line 883, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/jinja2/environment.py", line 857, in _load_template
    template = self.loader.load(self, name, globals)
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/jinja2/loaders.py", line 127, in load
    code = environment.compile(source, name, filename)
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/jinja2/environment.py", line 638, in compile
    self.handle_exception(source=source_hint)
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/Users/user/PycharmProjects/microblog/app/templates/index.html", line 20, in template
    {% endblock %}
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'endblock'. You probably made a nesting mistake. Jinja is expecting this tag, but currently looking for 'elif' or 'else' or 'endif'. The innermost block that needs to be closed is 'if'.
2020-11-17 17:54:18,377 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:43]
2020-11-17 18:52:57,943 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:43]
2020-11-19 18:01:56,718 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:45]
2020-11-19 18:10:06,826 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:45]
2020-11-19 20:26:54,831 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:45]
2020-11-20 00:45:36,870 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:45]
2020-11-20 23:20:22,633 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:47]
2020-11-27 00:41:09,347 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:49]
2020-12-01 23:23:19,928 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-01 23:29:46,778 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-03 00:46:29,995 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-03 00:51:53,521 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-03 01:26:34,046 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-03 22:19:35,028 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-03 22:20:35,530 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-04 00:13:20,715 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-04 19:59:41,216 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-04 20:02:52,769 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-04 20:03:18,771 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-04 20:08:19,696 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-04 20:08:56,646 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-04 20:10:21,216 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-04 20:10:33,926 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-04 20:11:35,353 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-04 20:17:18,327 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-08 00:29:16,024 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-08 00:29:39,396 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-08 00:29:47,946 INFO: Microblog startup [in /Users/user/PycharmProjects/microblog/app/__init__.py:53]
2020-12-08 00:31:25,005 ERROR: Exception on /user/yulia [GET] [in /Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/Users/user/PycharmProjects/microblog/app/routes.py", line 146, in user
    return render_template('user.html', user=user, posts=posts.items,
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/templating.py", line 137, in render_template
    return _render(
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/Users/user/PycharmProjects/microblog/app/templates/user.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/Users/user/PycharmProjects/microblog/app/templates/base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 1, in top-level template code
    {% block doc -%}
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 4, in block "doc"
    {%- block html %}
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 20, in block "html"
    {% block body -%}
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 23, in block "body"
    {% block content -%}
  File "/Users/user/PycharmProjects/microblog/app/templates/base.html", line 49, in block "content"
    {% block app_content %}{% endblock %}
  File "/Users/user/PycharmProjects/microblog/app/templates/user.html", line 26, in block "app_content"
    {{ form.hidden_tag() }}
  File "/Users/user/PycharmProjects/venvs/microblog/lib/python3.8/site-packages/jinja2/environment.py", line 471, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'form' is undefined
